File: q1897/q1897.py
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        charMap = defaultdict(int)
        for word in words: 
            for char in word: 
                charMap[char] += 1
        
        expectedCount = len(words)
        for count in charMap.values(): 
            if count % expectedCount != 0: 
                return False
        return True

File: q1897/q1897.java
class Solution {
    public boolean makeEqual(String[] words) {
        Map<Character, Integer> charMap = new HashMap<>();
        for (String word : words) {
            for (int i = 0; i < word.length(); i++) {
                char c = word.charAt(i);
                charMap.put(c, charMap.getOrDefault(c, 0) + 1);
            }
        }

        int expectedCount = words.length;
        for (int count : charMap.values()) {
            if (count % expectedCount != 0)
                return false;
        }
        return true;
    }
}